src/main/java/uk/gov/hmcts/darts/common/repository/ExternalObjectDirectoryRepository.java

```diff
@@ -466,4 +476,36 @@ AND eod.externalLocationType.id IN (1, 2)
                 AND eod.externalLocationType.id IN (1, 2)
         """)
     List<ExternalObjectDirectoryEntity> findStoredInInboundAndUnstructuredByTranscriptionId(@Param("transcriptionDocumentId") Integer id);
+
+    @Query(value = """
+        select fileSize from
+        (
+            (
+                select file_size as fileSize from darts.media as med
+                join darts.external_object_directory as eod on eod.med_id = med.med_id
+                where eod.eod_id = :externalObjectDirectoryId
+            )
+            union
+            (
+                select file_size as fileSize from darts.annotation_document as ado
+                join darts.external_object_directory as eod on eod.ado_id = ado.ado_id
+                where eod.eod_id = :externalObjectDirectoryId
+            )
+            union
+            (
+                select file_size as fileSize from darts.case_document as cad
+                join darts.external_object_directory as eod on eod.cad_id = cad.cad_id
+                where eod.eod_id = :externalObjectDirectoryId
+            )
+            union
+            (
+                select file_size as fileSize from darts.transcription_document as trd
+                join darts.external_object_directory as eod on eod.trd_id = trd.trd_id
+                where eod.eod_id = :externalObjectDirectoryId
+            )
+        ) a
+        """, nativeQuery = true)
+    Long findFileSize(Integer externalObjectDirectoryId);
```

Comment:
could we do this in code since we have already the Eod entity we can use?
If lazy loading works we could navigate from there to the not null associated object (media, transcript...), otherwise we could just fetch the associated object from the repo and get its size?
This way we can avoid this complex native very specific query

Reply:
This was considered the most optimal query wise. If we add @Transactional it will implode the amount of queries called.

Reply:
i'm not sure whats best here. I'm not the biggest fan of native queries, but in this scenario, it means 1 query can be used instead of 4, so it is more performant. 
Its a question i have asked before, but no real guidance.

Reply:
Im not sure if UNION is performant. I am inclined towards @mario-paniccia solution i.e. achieving this via entity association rather than native query with UNION @cmagowan-hmcts. Happy to discuss it further.

Reply:
with my suggested solution it's going to be only one query to get by id one of media, transcript, annotation or case document. Since we already have the EOD entity